src/librustc_resolve/lib.rs

```diff
@@ -4796,10 +4796,18 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
                 err.span_suggestion_with_applicability(
                     binding.span,
                     rename_msg,
-                    if snippet.ends_with(';') {
-                        format!("{} as {};", &snippet[..snippet.len() - 1], suggested_name)
+                    if snippet.contains(" as ") {
+                        format!(
+                            "{} as {}",
+                            &snippet[..snippet.find(" as ").unwrap()],
```

Comment:
Given that we already have a new name being given, let's say `use foo as bar;`, then we should probably suggest `use foo as other_bar;`.

Reply:
That's what is happening